### PR TITLE
Updates to allow building on macOS

### PR DESCRIPTION
**Pull request by mathomp4**
Per @tclune, building with gfortran on macOS has some issues. Following
his lead with pFUnit, I've created a new file, `mojave_fix.h` and then
added the header to the C files that needed it.

Note, it looks like it has to be the *FIRST* header in a file that needs
it. Ugly, but what can you do.

### CODE REVIEW

**Comment by mathomp4** [comment]
I am assuming this is zero-diff and shouldn't have issues on Linux due to its use in pFUnit. ==> Testing now.

**Comment by mathomp4** [comment]
I'm blocking this until I can talk with @tclune

**Review by tclune — DISMISSED** [review]
Matt and I talked.   I defer to him as to whether to wait for a fix in MOM so that it can get NetCDF include files directly instead of through FMS.

**Comment by mathomp4** [comment]
Okay. This is a good thing. But, I'm keeping the blocker on until the MOM change (GEOS-ESM/MOM5#8) is approved. That needs to be there for this to proceed.

**Comment by mathomp4** [comment]
As GEOS-ESM/MOM#8 might not be approved for a while, I'm going to split this PR into two. One for the needed macOS changes and one for the `PRIVATE` of the includes. `mojave_fix.h` is specific enough that nothing else in GEOS will accidentally `#include` it. If so, we'd make it more generic and put in something like MAPL.

**Review by tclune — APPROVED** [review]

**Pull request merged**
